front/app/utils/cl-intl/useIntl.tsx

```diff
@@ -1,16 +1,27 @@
-import { useCallback } from 'react';
+import { useCallback, useState, useEffect } from 'react';
 import {
   // eslint-disable-next-line no-restricted-imports
   useIntl as useOriginalUseIntl,
   MessageDescriptor,
 } from 'react-intl';
 import { isNilOrError } from 'utils/helperUtils';
 import useAppConfiguration from 'api/app_configuration/useAppConfiguration';
-import useLocalize from 'hooks/useLocalize';
+import { localeStream } from 'utils/locale';
+import { getLocalizedWithFallback } from 'utils/i18n';
+import { Locale } from 'typings';
 
 const useIntl = () => {
   const intl = useOriginalUseIntl();
-  const localize = useLocalize();
+  const [locale, setLocale] = useState<Locale | undefined>(undefined);
```

Comment:
Why was it necessary to move this stuff form `useLocale` into here?

Reply:
Because for the multilingual preview to work now with all the components and hooks (`useLocalize`, `useLocale`, etc.) I need them to use the language from the language context and not the one from the platform. That's why I moved the logic around a little bit where the platform locale is set in the most high level language provider but we can have additional language providers below that one with a different language and all the components nested underneeth will have all of their `intl` logic affected by the closest one.

Reply:
I don't fully understand, I thought this was all already covered by using the `ContentBuilderLanguageProvider`. But it's fine, as long as you tested everything well

Reply:
Before it worked fine for the copy but not for `useLocale` and `useLocalize` which was not an issue for the report builder but now it became one with the homepage preview.